README.md

```diff
@@ -152,15 +152,18 @@ Description of an error or warning which occurred during parsing of the API desc
 - type (enum[string])
     - `error`
     - `warning`
-- component (enum[string]) - In which component of the compilation process the annotation occurred.
+- component (enum[string]) - In which component of the compilation process the annotation occurred
     - `apiDescriptionParser`
     - `parametersValidation`
     - `uriTemplateExpansion`
-- message (string) - Textual annotation. This is – in most cases – a human-readable message to be displayed to user.
-- location (array) - Locations of the annotation in the source file. A series of character-blocks, which may be non-continuous. For further details refer to API Elements' [Source Map](source-map) element.
-    - (array, fixed) - Continuous characters block. A pair of character index and character count.
-        - (number) - Zero-based index of a character in the source document.
-        - (number) - Count of characters starting from the character index.
+- message (string) - Textual annotation. This is – in most cases – a human-readable message to be displayed to user
+- location (array, fixed, nullable) - Location of the annotation in the source file, represented by a single range of line and column number pairs if available, or by `null` otherwise
```

Comment:
Since your changing the design of this, one thing to consider is we do have multiple-file support on our roadmap (where source maps could be against another file or span across multiple files).

That's why they are array of arrays in API Elements so we can add additional fields without breaking the structure too much.

Reply:
That's an interesting point, thanks. I'm probably fine leaving this sub-optimized for this case and to redesign it again once it happens. While this is a separate and public library, I don't think there's anything else than Dredd integrating on top of it, so breaking changes are cheap.

Reply:
Also, multiple-file support might need some redesign work in Dredd as well :)